controllers/integrationpipeline/integrationpipeline_adapter.go

```diff
@@ -189,6 +190,41 @@ func (a *Adapter) EnsureStatusReported() (controller.OperationResult, error) {
 	return controller.ContinueProcessing()
 }
 
+// EnsureStatusReportedInSnapshot will ensure that status of the integration test pipelines is reported to snapshot
+// to be consumed by user
+func (a *Adapter) EnsureStatusReportedInSnapshot() (controller.OperationResult, error) {
```

Comment:
is possible to merge this Ensure function with another ensure function and save one call to get status.

I'm not sure if it's worth, this is simple and easy to read

Reply:
Im for current state, as you said better readability.